config/initializers/content_security_policy.rb

```diff
@@ -4,23 +4,82 @@
 # For further information see the following documentation
 # https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Content-Security-Policy
 
-Rails.application.configure do
-  config.content_security_policy do |policy|
-    policy.default_src :self, :https
-    policy.font_src    :self, :https, :data
-    policy.img_src     :self, :https, :data
-    policy.object_src  :none
-    policy.script_src  :self, :https
-    policy.style_src   :self, :https
-    # Specify URI for violation reports
-    # policy.report_uri "/csp-violation-report-endpoint"
-  end
+# See https://developer.mozilla.org/en-US/docs/Web/HTTP/CSP for more CSP info.
+#
+# The resulting policy should be checked with:
+#
+# - https://csp-evaluator.withgoogle.com
+# - https://cspvalidator.org
+
+GOVUK_DOMAINS = [
+  '*.education.gov.uk',
+].uniq.freeze
+
+GOOGLE_ANALYTICS_DOMAINS = %w[www.google-analytics.com
+                              ssl.google-analytics.com
+                              stats.g.doubleclick.net
+                              www.googletagmanager.com
+                              *.ytimg.com
+                              www.youtube.com
+                              www.youtube-nocookie.com
+                              *.hotjar.com].freeze
+
+OPTIMIZE_DOMAINS = %w[www.googleoptimize.com
+                      optimize.google.com
+                      fonts.googleapis.com].freeze
 
-  # Generate session nonces for permitted importmap and inline scripts
-  config.content_security_policy_nonce_generator = ->(request) { request.session.id.to_s }
-  # config.content_security_policy_nonce_directives = %w(script-src)
+GOOGLE_STATIC_DOMAINS = %w[fonts.gstatic.com www.gstatic.com *.hotjar.com].freeze
 
-  # Report CSP violations to a specified URI. See:
-  # https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Content-Security-Policy-Report-Only
-  # config.content_security_policy_report_only = true
+Rails.application.config.content_security_policy do |policy|
+  policy.default_src :self, :https, *GOVUK_DOMAINS
+  policy.font_src    :self, :https, *GOVUK_DOMAINS, *GOOGLE_STATIC_DOMAINS, :data
+  policy.frame_src   :self, *GOOGLE_ANALYTICS_DOMAINS, *OPTIMIZE_DOMAINS
+  policy.img_src     :self,
+                     *GOVUK_DOMAINS,
+                     *GOOGLE_ANALYTICS_DOMAINS, # Tracking pixels
+                     *OPTIMIZE_DOMAINS,
+                     :data # Base64 encoded images
+  policy.object_src  :none
+  policy.script_src  :self,
```

Comment:
Am I right in thinking this will address these warnings?
<img width="622" alt="Screenshot 2022-05-16 at 10 16 07" src="https://user-images.githubusercontent.com/3261/168560312-e2064c33-2167-4a82-b11d-a8a3338306ae.png">

Reply:
Yes this will